libvast/src/system/index.cpp

```diff
@@ -428,143 +541,182 @@ caf::behavior index(caf::stateful_actor<index_state>* self, const path& dir,
   // simply waits for a worker).
   self->set_default_handler(caf::skip);
   self->state.has_worker.assign(
-    [=](expression& expr) {
-      auto respond = [&](auto&&... xs) {
-        auto mid = self->current_message_id();
-        unsafe_response(self, self->current_sender(), {}, mid.response_id(),
+    [=](caf::stream<table_slice_ptr> in) {
+      VAST_DEBUG(self, "got a new table slice stream");
+      return self->state.stage->add_inbound_path(in);
+    },
+    // The partition delegates the actual writing to the filesystem actor,
+    // so we dont really get more information than a binary ok/not-ok here.
+    [=](caf::result<atom::ok> write_result) {
+      if (write_result.err)
+        VAST_ERROR(self, "could not persist:", write_result.err);
+      else
+        VAST_VERBOSE(self, "successfully persisted partition");
+    },
+    // Query handling
+    [=](vast::expression expr) {
+      auto& st = self->state;
+      auto mid = self->current_message_id();
+      auto sender = self->current_sender();
+      auto client = caf::actor_cast<caf::actor>(sender);
+      auto respond = [=](auto&&... xs) {
```

Comment:
Do we still need this hack?

Reply:
As far as I understand, the purpose of this is to "respond" to the original sender without returning from the function. There's probably a better way to do it, but I couldnt immediately see how.